tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -131,32 +159,51 @@ public int PartialFlushMinSpans
         /// </summary>
         internal TransportType MetricsTransport { get; set; }
 
-        private void ConfigureMetricsTransport(IConfigurationSource source, bool isWindows)
+        private void ConfigureMetricsTransport(IConfigurationSource source)
         {
-            var metricsTransport = TransportType.UDP; // default
+            TransportType? metricsTransport = null;
 
             var dogStatsdPort = source?.GetInt32(ConfigurationKeys.DogStatsdPort);
 
-            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
-            // The agent will fail to start if it can not bind a port
-            if (dogStatsdPort == 0)
-            {
-                MetricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
+            MetricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
 
-                if (MetricsPipeName != null)
+            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict.
+            // The agent will fail to start if it can not bind a port.
+            // If the dogstatsd port isn't explicitly configured, check for pipes or sockets.
+            if (dogStatsdPort == 0 || dogStatsdPort == null)
+            {
+                if (!string.IsNullOrWhiteSpace(MetricsPipeName))
                 {
                     metricsTransport = TransportType.NamedPipe;
                 }
+                else
+                {
+                    // Check for UDS
+                    var metricsUnixDomainSocketPath = source?.GetString(ConfigurationKeys.MetricsUnixDomainSocketPath);
+                    if (metricsUnixDomainSocketPath != null)
+                    {
+                        metricsTransport = TransportType.UDS;
```

Comment:
Should we test File.Exists as for the default one below?

Reply:
I found in tests that depending on a startup race, the tracer can load here before the agent has a chance to create the socket. If we fail out, then at worst the application crashes, at best, we never send traces when we could do so.
Particularly in the docker container sample I've included, it wouldn't work at all if we failed because the file was missing.
I decided to honor the explicit config, and only check for existence of the file if it's conventions based. This is per the RFC.

Reply:
Ok. Can you maybe just add a comment here expliciting this.
Also it does mean that we could have the same problem by using the default (else below). I don't think we should wait or implement retries, so in the case where we use the defaults, we should maybe add a debug log stating that the socket wasn't created, wdyt?

Reply:
If the file doesn't exist, then that would mean we use the default TCP configuration which is `localhost:8126`
The existence of the file (and lack of explicit config) is what dictates it should be used.

I will add notes explaining this though.

Reply:
I added substantial test coverage in `ExporterSettingsTests`, will that do or should we also add comments here?